src/components/ArrowKeyFocusManager.js

```diff
@@ -28,91 +28,95 @@ const defaultProps = {
     shouldExcludeTextAreaNodes: true,
 };
 
-class ArrowKeyFocusManager extends Component {
-    componentDidMount() {
+function ArrowKeyFocusManager(props) {
+    const onArrowUpKeyRef = useRef();
+    const onArrowDownKeyRef = useRef();
+    const mounted = useRef(false);
+    const prevMaxIndex = useRef();
+
+    useEffect(() => {
+        onArrowUpKeyRef.current = () => {
+            if (props.maxIndex < 0) {
+                return;
+            }
+
+            const currentFocusedIndex = props.focusedIndex > 0 ? props.focusedIndex - 1 : props.maxIndex;
+            let newFocusedIndex = currentFocusedIndex;
+
+            while (props.disabledIndexes.includes(newFocusedIndex)) {
+                newFocusedIndex = newFocusedIndex > 0 ? newFocusedIndex - 1 : props.maxIndex;
+                if (newFocusedIndex === currentFocusedIndex) {
+                    // all indexes are disabled
+                    return; // no-op
+                }
+            }
+            props.onFocusedIndexChanged(newFocusedIndex);
+        };
+
+        onArrowDownKeyRef.current = () => {
+            if (props.maxIndex < 0) {
+                return;
+            }
+
+            const currentFocusedIndex = props.focusedIndex < props.maxIndex ? props.focusedIndex + 1 : 0;
+            let newFocusedIndex = currentFocusedIndex;
+
+            while (props.disabledIndexes.includes(newFocusedIndex)) {
+                newFocusedIndex = newFocusedIndex < props.maxIndex ? newFocusedIndex + 1 : 0;
+                if (newFocusedIndex === currentFocusedIndex) {
+                    // all indexes are disabled
+                    return; // no-op
+                }
+            }
+            props.onFocusedIndexChanged(newFocusedIndex);
+        };
+    });
+
+    const onArrowUpKey = useCallback(() => onArrowUpKeyRef.current(), []);
+    const onArrowDownKey = useCallback(() => onArrowDownKeyRef.current(), []);
```

Comment:
The way we set the key handlers is a bit weird, but I think it is necessary here, if we want to avoid unnecessary re-renders or resubscribing to the `KeyboardShortcut` inside `useEffect` below.

I am trying to accomplish something akin to the `useEvent` hook from [this RFC](https://github.com/reactjs/rfcs/blob/useevent/text/0000-useevent.md)